core/src/main/resources/error/error-classes.json

```diff
@@ -294,6 +300,18 @@
       "UDF class <className> doesn't implement any UDF interface"
     ]
   },
+  "NULLABLE_ARRAY_OR_MAP_ELEMENT" : {
+    "message" : [
+      "Array or map at <columnPath> contains nullable element while it's required to be non-nullable."
+    ],
+    "sqlState" : "42000"
+  },
+  "NULLABLE_COLUMN_OR_FIELD" : {
+    "message" : [
+      "Column or field <name> is nullable while it's required to be non-nullable."
```

Comment:
@cloud-fan will we report the column path in the error here ? or its going to be the name ?

Reply:
the `<name>` will be a fully qualified name, which includes the column path.

Reply:
See test https://github.com/apache/spark/pull/37011/files#diff-115433d6399124014f2600556a5f6ced9b79b64e4584c19d51997c3eff9e6f33R178